.github/workflows/notify-stalled-snapshots.yml

```diff
@@ -0,0 +1,55 @@
+---
+name: notify-stalled-snapshots
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 1 * * 1'
+
+permissions:
+  contents: read
+
+jobs:
+  filter:
+    runs-on: ubuntu-latest
+    timeout-minutes: 1
+    outputs:
+      matrix: ${{ steps.generator.outputs.matrix }}
+    steps:
+      - id: generator
+        uses: elastic/apm-pipeline-library/.github/actions/elastic-stack-snapshot-branches@current
+
+  notify:
+    runs-on: ubuntu-latest
+    needs: [filter]
+    strategy:
+      matrix: ${{ fromJson(needs.filter.outputs.matrix) }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          ref: "${{ matrix.branch }}"
+
+      - id: search
+        run: |-
+          if git --no-pager \
+            log --pretty=format: \
+              --name-only \
+              --since="7 days ago" \
+            | grep 'testing/environments/snapshot.yml' ; then
+            echo "found=false" >> $GITHUB_OUTPUT
+          else
+            echo "found=true" >> $GITHUB_OUTPUT
+          fi
+        env:
+          GH_TOKEN: ${{ github.token }}
+
+      - if: ${{ contains(steps.search.outputs.found, 'true') }}
+        name: Report obsoleted branches
+        uses: elastic/apm-pipeline-library/.github/actions/slack-message@current
```

Comment:
As far as I see there is no native support for email at the moment with GitHub actions, let's use slack notifications meanwhile.

Reply:
Slack is fine, thanks!